kingfisher_scrapy/spidermiddlewares.py

```diff
@@ -208,3 +211,28 @@ def process_spider_output(self, response, result, spider):
             item['data'].close()
             item['data'] = data
             yield item
+
+
+class RetryDataErrorMiddleware:
+    """
+    # https://docs.scrapy.org/en/latest/topics/spider-middleware.html#scrapy.spidermiddlewares.SpiderMiddleware.process_spider_exception
+    This method is called when a spider or process_spider_output() method (from a previous spider middleware) raises
+    an exception. Currently, if a BadZipFile exception is received, it retries the request up to 3 times per URL,
+    otherwise returns None for scrapy continuing processing the exception.
+    """
+    retries = defaultdict(int)
+
+    def process_spider_exception(self, response, exception, spider):
+        if isinstance(exception, BadZipFile):
+            if self.retries[response.request.url] > 3:
```

Comment:
Otherwise, we're trying 4 times.

```suggestion
            if self.retries[response.request.url] >= 3:
```